bika/lims/browser/clientfolder.py

```diff
@@ -96,93 +107,55 @@ def __init__(self, context, request):
-            if not mtool.checkPermission(ManageAnalysisRequests, client):
-                continue
-            clients.append(brain)
-
-        return clients
-
-    def folderitems(self):
-        self.contentsMethod = self.getClientList
-        items = BikaListingView.folderitems(self)
         registry = getUtility(IRegistry)
-        if 'bika.lims.client.default_landing_page' in registry:
-            landing_page = registry['bika.lims.client.default_landing_page']
-        else:
-            landing_page = 'analysisrequests'
-
-        for item in items:
-            if "obj" not in item:
-                continue
-            obj = item['obj']
+        self.landing_page = registry.get(self._LANDING_PAGE_REGISTRY_KEY,
+                                         'analysisrequests').encode('ascii')
 
-            item['replace']['title'] = "<a href='%s/%s'>%s</a>" % \
-                (item['url'], landing_page.encode('ascii'), item['title'])
-
-            item['EmailAddress'] = obj.getEmailAddress()
-            item['replace']['EmailAddress'] = "<a href='%s'>%s</a>" % \
-                ('mailto:%s' % obj.getEmailAddress(), obj.getEmailAddress())
-            item['Phone'] = obj.getPhone()
-            item['Fax'] = obj.getFax()
-            item['ClientID'] = obj.getClientID()
+        return super(ClientFolderContentsView, self).__call__()
 
-        return items
+    def isItemAllowed(self, obj):
+        """
+        Returns true if the current user has Manage AR rights for the current
+        Client (item) to be rendered.
+        :param obj: client to be rendered as a row in the list
+        :type obj: ATContentType/DexterityContentType
+        :return: True if the current user can see this Client. Otherwise, False.
+        :rtype: bool
+        """
+        mtool = api.get_tool('portal_membership')
+        client_object = api.get_object(obj)
+        return mtool.checkPermission(ManageAnalysisRequests, client_object)
+
+    def folderitem(self, obj, item, index):
+        """
+        Applies new properties to the item (Client) that is currently being
+        rendered as a row in the list
+        :param obj: client to be rendered as a row in the list
+        :param item: dict representation of the client, suitable for the list
+        :param index: current position of the item within the list
+        :type obj: ATContentType/DexterityContentType
+        :type item: dict
+        :type index: int
+        :return: the dict representation of the item
+        :rtype: dict
+        """
+        client_object = api.get_object(obj)
+        item['replace']['title'] = "<a href='%s/%s'>%s</a>" % \
+                                   (item['url'], self.landing_page,
+                                    item['title'])
+        email = client_object.getEmailAddress()
+        item['replace']['EmailAddress'] = "<a href='%s'>%s</a>" % \
+                                          ('mailto:%s' % email, email)
```

Comment:
Better to test and maintain would be to have here explicit methods, e.g.: `get_client_email_link(client, **kw)` and `get_landing_page_url(**kw)` and pass in the `item` as the keyword args `**item`

Reply:
Indeed. I can add the function `get_email_link` in `bika.lims.utils`. Re `landing_page` thing, I think this is the only place where is used, so not sure if adding an explicit method for this is reasonable.